File: Device/DeviceController.Data/IrrigationProgram.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DeviceController.Data
{
    public class IrrigationProgram
    {
        public int Id;
        public string Name;
        public DateTime Start;
        public DateTime Finished;
        public int Duration;
        public int DeviceId;
        public int SolenoidId;        
        public bool RequiresPump;        
    }
}

File: MobileApp/src/app/models/irrigation-program.ts
export class IrrigationProgram {
    id: number;
    name: string;
    start: Date;
    finished: Date;
    duration: number;
    solenoidId: number;
    requiresPump: boolean;
    deviceId: number;
    createdAt: Date;
    updatedAt: Date;
}
